cmd/nginx/flags.go

```diff
@@ -188,25 +192,47 @@ Feature backed by OpenResty Lua libraries.`)
 
 	parser.AnnotationsPrefix = *annotationsPrefix
 
-	// check port collisions
-	if !ing_net.IsPortAvailable(*httpPort) {
-		return false, nil, fmt.Errorf("Port %v is already in use. Please check the flag --http-port", *httpPort)
-	}
+	bindIpAddresses := make([]string, 0)
+	for _, ipAddressStr := range strings.Split(*bindAddresses, ",") {
+		ipAddressSanitized := strings.Replace(strings.Replace(ipAddressStr, "]", "", -1),
+			"[", "", -1)
 
-	if !ing_net.IsPortAvailable(*httpsPort) {
-		return false, nil, fmt.Errorf("Port %v is already in use. Please check the flag --https-port", *httpsPort)
+		ip := net.ParseIP(ipAddressSanitized)
+		if ip == nil {
+			return false, nil, fmt.Errorf("string %v is not an ip address", ipAddressStr)
+		}
+		if !ing_net.IsIPv6Enabled() && ing_net.IsIPV6(ip) {
+			glog.Warningf("IPv6 is disabled, skipping IPv6 address: %s", ipAddressStr)
+		} else {
+			bindIpAddresses = append(bindIpAddresses, ipAddressStr)
+		}
 	}
+	// check port collisions
+	for _, ipAddress := range bindIpAddresses {
+		if !ing_net.IsPortAvailable(ipAddress, *httpPort) {
+			return false, nil, fmt.Errorf("port %v:%v is already in use. Please check the flag --http-port",
```

Comment:
(nit) Printed errors should start with a capital letter, and you could use %d to make it clear you're printing an integer.